tests/unit/test_adapter.py

```diff
@@ -0,0 +1,19 @@
+from openscm.adapter import Adapter
+import pytest
+
+def test_adapter():
```

Comment:
This test is almost just a stub as the class is almost abstract.
What do you mean by "more easily scalable"?

Reply:
> What do you mean by "more easily scalable"?

Write a base testing class which the user then subclasses and as a result all the 'basic' tests are run with their adapter too.

Reply:
Instead of requiring subclassing a test class we could also just apply a standard test to each individual adapter.

Reply:
That is possible but I don't think it scales very well. Mainly because it then becomes difficult to easily specify that you want to run the tests for a given model only (whereas if you have subclasses you can do e.g. `pytest tests/adapters/test_adapters.py::TestMAGICCAdapter) and because it makes it hard for users to implement an adapter outside of OpenSCM (which may happen at some point). I suspect it also makes it impossible for implementers to add extra one or two steps before/after the core test for things which are specific to their model for that test without having to re-write the whole test. I've done this subclassing before (see [here](https://github.com/znicholls/netcdf-scm/blob/master/tests/unit/test_scmcube.py)) and started mucking around with possible implementations for OpenSCM [here](https://github.com/openclimatedata/openscm/blob/scmcallib/tests/integration/test_adapters.py) so whilst it is more complicated to setup it does ensure that every adapter goes through the tests we want. Another reason I suspect this is the right approach is that TestClass subclassing is how pandas has setup their tests for extension arrays (see [here](https://pandas.pydata.org/pandas-docs/stable/development/extending.html#testing-extension-arrays)).